ui/src/views/infra/zone/ZoneWizardRegisterTemplate.vue

```diff
@@ -0,0 +1,339 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+<template>
+    <div>
+      <a-table
+        :columns="columns"
+        :dataSource="predefinedTemplates"
+        :rowSelection="rowSelection"
+        :loading="loading"
+        :scroll="{ y: 450 }"
+        size="middle"
+        :rowKey="record => record.id"
+        :pagination="false"
+        class="form-content"
+      >
+      <template #bodyCell="{ column, record }">
+          <template v-if="column.key === 'name'">
+            <os-logo :osName="record.name" size="xl" />
+            {{ record.name }}
+          </template>
+        </template>
+      </a-table>
+      <div class="form-action">
+        <a-button class="button-back" @click="handleDone">{{ selectedRowKeys.length > 0 ? $t('label.done') : $t('label.skip') }}</a-button>
+        <a-button class="button-next" type="primary" @click="handleSubmit" ref="submit">{{ $t('label.register.template') }}</a-button>
+      </div>
+    </div>
+  </template>
+
+<script>
+import { api } from '@/api'
+import { genericCompare } from '@/utils/sort.js'
+import OsLogo from '@/components/widgets/OsLogo'
+
+export default {
+  name: 'ZoneWizardRegisterTemplate',
+  components: {
+    OsLogo
+  },
+  props: {
+    zoneid: {
+      type: String,
+      required: false
+    }
+  },
+  data: () => ({
+    columns: null,
+    loading: false,
+    predefinedTemplates: null,
+    rowKey: 0,
+    selectedRowKeys: [],
+    defaultOsTypeId: null,
+    deployedTemplates: {}
+  }),
+  created () {
+    this.initForm()
+  },
+  computed: {
+    rowSelection () {
+      return {
+        selectedRowKeys: this.selectedRowKeys || [],
+        onChange: this.onSelectRow,
+        getCheckboxProps: (record) => {
+          return {
+            disabled: record.disabled
+          }
+        }
+      }
+    }
+  },
+  methods: {
+    async initForm () {
+      this.columns = [
+        {
+          title: 'Name',
+          dataIndex: 'name',
+          key: 'name',
+          width: 240,
+          sorter: (a, b) => { return genericCompare(a.name || '', b.name || '') }
+        },
+        {
+          title: 'Arch',
+          dataIndex: 'arch',
+          key: 'arch',
+          width: 80,
+          sorter: (a, b) => { return genericCompare(a.arch || '', b.arch || '') }
+        },
+        {
+          title: 'URL',
+          dataIndex: 'url',
+          key: 'url'
+        }
+      ]
+      this.predefinedTemplates = [
+        {
+          id: 1,
+          name: 'Ubuntu 24.04',
+          version: '24.04',
+          osCategoryName: 'Ubuntu',
+          arch: 'x86_64',
+          url: 'https://cloud-images.ubuntu.com/releases/noble/release/ubuntu-24.04-server-cloudimg-amd64.img'
```

Comment:
@sudo87 should we hardcode the URLs, or is there a way to discover the latest LTS urls somehow?

Reply:
correct
moreover, some images might be moved in the future